ballerina/client.bal

```diff
@@ -125,16 +119,36 @@ public type Options record {|
 type ClientConfiguration record {|
     # The JDBC URL to be used for the database connection.
     string? url;
-    # If the database is secured, the username.
-    string? user;
-    # The password of the database associated with the provided username
-    string? password;
+    # The authentication configuration for the Snowflake client.
+    AuthConfig authConfig;
     # The JDBC client properties
     Options? options;
     # The `sql:ConnectionPool` to be used for the connection. If there is no `connectionPool` provided, the global connection pool (shared by all clients) will be used
     sql:ConnectionPool? connectionPool;
 |};
 
+# Represents the authentication configuration for the Snowflake client.
+public type AuthConfig BasicAuth|KeyBasedAuth;
+
+# Represents the basic authentication configuration for the Snowflake client.
+public type BasicAuth record {
+    # The username of the Snowflake account
+    string user;
+    # The password of the Snowflake account
+    string password;
+};
+
+# Represents the key-based authentication configuration for the Snowflake client.
+public type KeyBasedAuth record {
+    # The username of the Snowflake account
+    string user;
```

Comment:
Do the other auth types requires the common user field?

Reply:
I am not sure about that. since the user is a part of user credentials, I kept it inside the authconfigs